### PR TITLE
add helper function to convert chirrtl to firrtl.

### PR DESCRIPTION
**Pull request by sequencer**
We purge Driver too radically, thus lost some API, this PR add API to convert chirrtl to firrtl.

### Contributor Checklist

- [x] Did you add Scaladoc to every public function/method?
- [x] Did you add at least one test demonstrating the PR?
- [x] Did you delete any extraneous printlns/debugging code?
- [x] Did you specify the type of improvement?
- [ ] Did you add appropriate documentation in `docs/src`?
- [x] Did you state the API impact?
- [x] Did you specify the code generation impact?
- [x] Did you request a desired merge strategy?
- [x] Did you add text to be included in the Release Notes for this change?

#### Type of Improvement
- new feature/API

#### API Impact
add api chisel3.stage.convert(chirrtl: cir.Circuit): fir.Circuit

#### Backend Code Generation Impact
none

#### Desired Merge Strategy
- Rebase: You will rebase the PR onto master and it will be merged with a merge commit.

#### Release Notes
add api chisel3.stage.convert(chirrtl: cir.Circuit): fir.Circuit to convert cirrtl to firrtl

### Reviewer Checklist (only modified by reviewer)
- [ ] Did you add the appropriate labels?
- [ ] Did you mark the proper milestone (3.2.x, 3.3.x, 3.4.0, 3.5.0) ?
- [ ] Did you review?
- [ ] Did you check whether all relevant Contributor checkboxes have been checked?
- [ ] Did you mark as `Please Merge`?

### CODE REVIEW

**Comment by sequencer** [comment]
I'm tuning firesim, doc/test will be added later.

File: src/main/scala/chisel3/stage/ChiselStage.scala [hunk 1]
```diff
@@ -172,6 +172,23 @@ object ChiselStage {
       .get
   }
 
+  def convert(chirrtl: cir.Circuit): fir.Circuit = {
```
Comment (edwardcwang) [inline src/main/scala/chisel3/stage/ChiselStage.scala:175]: Scaladoc?

Comment (jackkoenig) [reply src/main/scala/chisel3/stage/ChiselStage.scala:175]: Yes ScalaDoc that also discourages using this

**Review by jackkoenig — COMMENTED** [review]
I'm okay with adding this back since `cir.Circuit` is public. If we were to deprecate that and then make that private then this can be deprecated and removed at the same time.

**Comment by sequencer** [comment]
need @jackkoenig  review to this PR :)

**Review by jackkoenig — APPROVED** [review]

**Comment by davidbiancolin** [comment]
This is probably a stupid question, but why doesn't emitFirrtl return a tuple of (String, Annotations), where _2 => elaboration-generated annotations? Most non-trivial chisel will generate some annotations right? 

 Is there a blessed way of doing this, instead of calling elaborate and then emit as we used to do with the old driver
 (here: https://github.com/firesim/firesim/blob/master/sim/midas/src/main/scala/midas/passes/SimulationMapping.scala#L96)? 
 
I'd be happy to PR a method, say `emitFirrtlAndElaboratedAnnotations`, to ChiselStage that would return a tuple if that makes sense.

**Comment by sequencer** [comment]
Oh, sorry for the delay response, I totally missed this.

> but why doesn't emitFirrtl return a tuple of (String, Annotations), where _2 => elaboration-generated annotations? Most non-trivial chisel will generate some annotations right?

Sounds really reasonable, but I'm thinking if the use case here is conflict to `NoRunFirrtlCompilerAnnotation`?
This maybe will introduce more codes to firesim.
I'll file a PR later.

**Pull request merged**